### PR TITLE
Mnist softmax is learning 

### PR DESCRIPTION
**Pull request by jburroni**
I think we can merge this

### CODE REVIEW

File: Packages/TensorFlow-Examples.pck.st [hunk 1]
```diff
@@ -804,25 +836,27 @@ initialize
...
+!MNIST3LayersNNExamplePlan methodsFor: 'initialization' stamp: 'JB 2/11/2017 14:21:08'!
 initializeInferenceGraph
 	| hidden1 hidden2 |
-	prediction _ graph fromBlock: [:image |
+	prediction _ graph fromBlock: [:image | | axis |
 		input _ image.
 		hidden1 _ (image * weights1 + biases1) rectified.
 		hidden2 _ (hidden1 * weights2 + biases2) rectified.
-		hidden2 * weights3 + biases3].
+		netInput _ hidden2 * weights3 + biases3.
```
Comment (gerasdf) [inline Packages/TensorFlow-Examples.pck.st]: why is it called netInput? it's roughly at then of the forward path, I'd intuitively say it's closer to being the output of the net, than an input...

**Comment by jburroni** [comment]
Because it's the net input for the activation function. That is the typical
name given in the literature. Logits makes sense only for sigmoid
activation

**Comment by jburroni** [comment]
Following with the biological model, it's the input to the neuron

**Comment by gerasdf** [comment]
That's what I thoght... I still don't think the literature is using a good
name... Lets keep it, but lets also think about it.

I don't think netInput is a good name because:

Net may also refer to the net (the neural net).
Input makes the reader think that it's an... input (to the system).

I understand this values are the net (pure) input to the following
operation (the activation). Maybe activationInput? unfoldedOutput?
unclassifiedOutput?

I also see that netInput is very deep in the literature... so... maybe...
oh well...

cheers!

File: Packages/TensorFlow-Examples.pck.st [hunk 2]
```diff
@@ -1090,30 +1090,30 @@ initializeInferenceGraph
...
+		named: 'learning'! !
+
+!MNISTSoftMaxExamplePlan methodsFor: 'initialization' stamp: 'gera 2/12/2017 23:09:05'!
 initializeLossGraph
 	loss _ graph
 		fromBlock: [:expected |
-			expectedForComparison _ expected.
-			(netInput sparseSoftmaxCrossEntropyWithLogits: expected) meanOn: #(0) asInt32Tensor]
-		inputTypes: {TF_Tensor typeInt32}! !
+			expectedLabel _ expected.
```
Comment (gerasdf) [inline Packages/TensorFlow-Examples.pck.st]: Javi, check how I unified part of the loss calculation path and the learning path.
Both now use the input (expectedLabel) and same activation node, the loss side uses output #0 from the softmax and the learning side uses output #1.

I believe it's still correct, but a bit simpler and a bit more aligned with the ideas of the softmax node, which has two outputs.

If you confirm it's good and like it, merge it :-)

:+1:
